pkg/controller/networkpolicy/controller_querier_replier.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package networkpolicy provides NetworkPolicyController implementation to manage
+// and synchronize the Pods and Namespaces affected by Network Policies and enforce
+// their rules.
+
+package networkpolicy
+
+import (
+	networkingv1beta1 "github.com/vmware-tanzu/antrea/pkg/apis/networking/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/apiserver/storage"
+	v1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+
+	antreatypes "github.com/vmware-tanzu/antrea/pkg/controller/types"
+)
+
+type EndpointQuerier interface {
+	QueryNetworkPolicies(namespace string, podName string) *EndpointQueryResponse
+}
+
+// EndpointQueryReplier is responsible for handling query requests from antctl query
+type EndpointQueryReplier struct {
+	// addressGroupStore is the storage where the populated Address Groups are stored.
+	addressGroupStore storage.Interface
+	// appliedToGroupStore is the storage where the populated AppliedTo Groups are stored.
+	appliedToGroupStore storage.Interface
+	// internalNetworkPolicyStore is the storage where the populated internal Network Policy are stored.
+	internalNetworkPolicyStore storage.Interface
+	// podInformer is used to check existence of a selected pod
+	podInformer coreinformers.PodInformer
+}
+
+// EndpointQueryResponse is the reply struct for QueryNetworkPolicies
+type EndpointQueryResponse struct {
```

Comment:
still puzzled as to why we don't need the json tags. I still think we should add them for consistency with the rest of the code and in case they need to be tweaked later.